pilot/pkg/proxy/envoy/v2/internalgen.go

```diff
@@ -79,27 +79,22 @@ func (sg *InternalGen) OnNack(node *model.Proxy, dr *discovery.DiscoveryRequest)
 	sg.startPush(TypeURLNACK, []*any.Any{util.MessageToAny(dr)})
 }
 
-// startPush is similar with DiscoveryServer.startPush() - but called directly,
-// since status discovery is not driven by config change events.
-// We also want connection events to be dispatched as soon as possible,
-// they may be consumed by other instances of Istiod to update internal state.
-func (sg *InternalGen) startPush(typeURL string, data []*any.Any) {
+// PushAll will immediately send a response to all connections that
+// are watching for the specific type.
+// TODO: additional filters can be added, for example namespace.
+func (s *DiscoveryServer) PushAll(res *discovery.DiscoveryResponse) {
```

Comment:
Why is this in DiscoveryServer? seems like it should be with the other experimental code in internalgen?